### PR TITLE
Add Slow CI reminder bot

### PR DESCRIPTION
**Pull request by ydshieh**
# What does this PR do?

@amyeroberts and @gante both mentioned that it would be helpful if we have such a reminder (so we won't forget to trigger whenever necessary)

p.s. 

- I am using `comment_bot_token` whose owner is [HuggingFaceDocBuilderDev](https://github.com/HuggingFaceDocBuilderDev), which might seems a bit strange. But I think it's fine (and not to create a new member with a new token again)

Screenshot 2024-09-16 115318

### CODE REVIEW

File: .github/workflows/slow_ci_remainder.yml
Comment (ydshieh) [inline .github/workflows/slow_ci_remainder.yml:5]: For security reason, see [https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#pull_request_target]

> This event runs in the context of the base of the pull request, rather than in the context of the merge commit, as the pull_request event does. This prevents execution of unsafe code from the head of the pull request that could alter your repository or steal any secrets you use in your workflow. This event allows your workflow to do things like label or comment on pull requests from forks. Avoid using this event if you need to build or run code from the pull request.

Comment (ydshieh) [reply .github/workflows/slow_ci_remainder.yml:5]: Also we need to use `pull_request_target` so every time a push to a PR will trigger the workflow without an approval by internal team members

> Note: Workflows triggered by pull_request_target events are run in the context of the base branch. Since the base branch is considered trusted, **workflows triggered by these events will always run, regardless of approval settings**. For more information about the pull_request_target event, see "[Events that trigger workflows](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request_target)."

**Review by gante — COMMENTED** [review]
Thank you for starting this PR! 🤗 

I have a few suggestions to improve the workflow from both ends (contributor and maintainer):
1. State in the text that this is a reminder for the maintainer, not for the contributor 🤗 
2. Add a link to our notion page, so that we can quickly confirm the commands to run the slow CI
3. Not sure if possible/easy, but it would be cool to skip this command if there are no relevant slow tests. At the very least, if no `.py` files were changed

**Comment by ydshieh** [comment]
> Thank you for starting this PR! 🤗
> 
> I have a few suggestions to improve the workflow from both ends (contributor and maintainer):
> 
>     1. State in the text that this is a reminder for the maintainer, not for the contributor 🤗
> 
>     2. Add a link to our notion page, so that we can quickly confirm the commands to run the slow CI
> 
>     3. Not sure if possible/easy, but it would be cool to skip this command if there are no relevant slow tests. At the very least, if no `.py` files were changed

Great suggestion, thank you!

**Comment by amyeroberts** [comment]
+1 to @gante points 2 & 3

For 1, I think we can put some of this on the contributors to enable (even if they're not responsible for ensuring they've run) to save maintainers some time :) 

We could add a message along the lines of: 
```
Before merging slow tests should be run. To enable this:
* Add the `run-slow` label to the PR
* When your PR is ready for merge and all reviewers' comments have been addressed, push an empty commit with the command `[run-slow]` followed by a comma separated list of all the models to be tested i.e. `[run_slow] model_to_test_1, model_to_test_2`
```

**Comment by ydshieh** [comment]
Hey! Comments addressed. I change a bit of wording (cc @amyeroberts). WDYT?

```
Before merging this pull request, slow tests CI should be triggered. To enable this:

    Add the run-slow label to the PR
    When your PR is ready for merge and all reviewers' comments have been addressed, push an empty commit with the command [run-slow] followed by a comma separated list of all the models to be tested, i.e. [run_slow] model_to_test_1, model_to_test_2
    A transformers maintainer will then approve the workflow to start the tests

(For maintainers) The documentation for slow tests CI on PRs is [here](https://www.notion.so/huggingface2/CI-for-pull-requests-8335bd9217d24d1e8ed1d3a016f39804).
```

**Comment by LysandreJik** [comment]
I think @ArthurZucker's approval would count more than mine on this

File: .github/workflows/slow_ci_remainder.yml
Comment (ArthurZucker) [inline .github/workflows/slow_ci_remainder.yml:7]: we can most probably filter modeling codes here

Comment (ArthurZucker) [reply .github/workflows/slow_ci_remainder.yml:7]: also not sure we want slow tests to run on ALL PR, otherwise we would just activate it by default

**Review by ArthurZucker — COMMENTED** [review]
Looks nice, would be better IMO if we don't have @ydshieh commenting, but a huggy-bot or something! 

I think we can filter a little bit if it's modeling changes to important models only / new models WDYT?

File: .github/workflows/slow_ci_remainder.yml
Comment (ydshieh) [reply .github/workflows/slow_ci_remainder.yml:7]: I would like to hear from @amyeroberts and @gante . But if I understand correctly, changes to generation related file or even soom test files, they both count.

We don't activate it by default is sometimes we don't want to run too many files in a PR slow CI.

And here it's just a remind for reviewers. If they think a PR is not worth a slow CI, we don't need to trigger it.

Comment (ydshieh) [reply .github/workflows/slow_ci_remainder.yml:7]: > if we don't have @ydshieh commenting, but a huggy-bot or something!

It's not me commenting 😆 , it would be [HuggingFaceDocBuilderDev](https://github.com/HuggingFaceDocBuilderDev),

Comment (amyeroberts) [reply .github/workflows/slow_ci_remainder.yml:7]: Yeah, it's a bit of a balancing act here -- just like the fetcher it's very hard to guarantee we'll always capture the minimal set of tests needed to be run and have to accept either some misses or too many selected. 

My preference would be: just trigger this warning for a subset of files we know pretty much ALWAYS need slow model tests: modeling code changes; changes to generation code; changes to attention mask creation. 

For all other cases, it's up to the discretion of the reviewer to ask for slow tests or not.

If there's a set of files we always run slow tests for, then we can always update this to include; or remove paths if it's annoying. That's the great thing about software!

Comment (ydshieh) [reply .github/workflows/slow_ci_remainder.yml]: @ArthurZucker 

changed it to

```
    paths:
      - "src/transformers/models/*/*.py"
      - "src/transformers/generation/*.py"
      - "src/transformers/pipelines/*.py"
      - "tests/models/*/*.py"
      - "tests/generation/*/*.py"
      - "tests/pipelines/*/*.py"
```

**Comment by ydshieh** [comment]
Any further comment @ArthurZucker ?

File: .github/workflows/slow_ci_remainder.yml
Comment (ArthurZucker) [inline .github/workflows/slow_ci_remainder.yml:52]:
```suggestion
          Hey! 🤗 Thanks for your contribution to the `transformers` library!
          Before merging this pull request, slow tests CI should be triggered. To enable this:
```

**Review by ArthurZucker — COMMENTED** [review]
Yeah a lot better thanks!
Can we have @glegendre01 's approval for the security of using outsourced actions?

**Comment by ydshieh** [comment]
> Yeah a lot better thanks! Can we have @glegendre01 's approval for the security of using outsourced actions?

That action is used in [huggingface/doc-builder](https://github.com/huggingface/doc-builder/blob/1b42854493622c0c8c31579a12078d6c71f5c9d3/.github/workflows/upload_pr_documentation.yml#L122)

```
      - name: Add doc comment if not present
        uses: thollander/actions-comment-pull-request@v2
        if: steps.find_comment.outputs.comment-id == ''

        with:
          message: 'The docs for this PR live [here](${{ steps.hfhub-context.outputs.hub_docs_url }}). All of your documentation changes will be reflected on that endpoint. The docs are available until 30 days after the last update.'
          pr_number: ${{ steps.github-context.outputs.pr_number }}
          GITHUB_TOKEN: ${{ secrets.comment_bot_token }}
```

**Review by ArthurZucker — APPROVED** [review]
Okay, then just this one ` peter-evans/find-comment@v2`! 🤗 LGTM otherwise thanks 🤗

**Comment by ydshieh** [comment]
oh, i forget that one, but it's also used already

https://github.com/huggingface/doc-builder/blob/1b42854493622c0c8c31579a12078d6c71f5c9d3/.github/workflows/upload_pr_documentation.yml#L115

(I won't keep silence when I actually a new action - you can trust me!)

**Comment by ydshieh** [comment]
I will wait and merge on Monday instead. Don't want to confuse the contributors during the weekend.

Thanks for review!

**Pull request merged**
